Controllers/Frontend/Transparent.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+use Shopware\Components\CSRFWhitelistAware;
+use Shopware\Components\Logger;
+
+// phpcs:ignore PSR1.Classes.ClassDeclaration.MissingNamespace, Squiz.Classes.ValidClassName.NotCamelCaps, Generic.Files.LineLength.TooLong
+class Shopware_Controllers_Frontend_Transparent extends Shopware_Controllers_Frontend_Payment implements CSRFWhitelistAware
+{
+
+    /**
+     * @var Logger
+     */
+    private $logger;
+
+    public function preDispatch()
+    {
+        $this->logger = $this->get('adyen_payment.logger');
+    }
+
+    public function getWhitelistedCSRFActions()
+    {
+        return ['redirect'];
+    }
+
+    /**
+     * Transparent redirect to solve 3DS1 issue same site cookie policy issue
+     * Used to fetch POST return data and redirect locally to allow session usage
+     * https://github.com/Adyen/adyen-shopware5/issues/72
+     */
+    public function redirectAction()
+    {
+        $redirectUrl = Shopware()->Router()->assemble([
+            'controller' => 'process',
+            'action' => 'return',
+        ]);
+
```

Comment:
I'd assign these values at first to the variable and then use it in the retrievePostParams instead of passing as a parameter and assigning to the var in 1 step. It's easier to read it.

```suggestion

$allowedParams = ['MD', 'PaRes'];

```

Reply:
CR adjustments made.
Added a small addition to escape the POST parameters in the template as well.

Reply:
Good one, forgot to mention the value escape on the frontend indeed +1